Scripts/Generator/Floor.cs

```diff
@@ -0,0 +1,37 @@
+using Godot;
+using System;
+
+public class Floor : StaticBody
+{
+    public Floor(Vector2 position, Vector2 dimensions, Material material)
+    {
+        Vector3 wallUp = Vector3.Up;
+        Mesh wallMesh = new CubeMesh();
+        MeshInstance meshInstance = new MeshInstance
+        {
+            Mesh = wallMesh
+        };
+
+        // make a duplicate of the shader so that can customize the `size` uniform
+        // might want to see if this costs anything? I don't think so though
+        // 5 lines of shader code shouldn't cost anything
+        ShaderMaterial dupMaterial = (ShaderMaterial)material.Duplicate();
+        dupMaterial.SetShaderParam("size", new Vector2(dimensions.x, dimensions.y));
+        meshInstance.SetSurfaceMaterial(0, dupMaterial);
+
+        CollisionShape collisionShape = new CollisionShape()
+        {
+            Shape = new BoxShape()
+        };
+        AddChild(meshInstance);
+        AddChild(collisionShape);
+
+        // divide scale by 2 because not sure why godot seems to double the effect
+        Scale = new Vector3(dimensions.x / 2, 1, dimensions.y / 2);
```

Comment:
Because CubeMesh and BoxShape is by default 2x2x2 at scale 1.
I suggest to do
```cs
Mesh floorMesh = new CubeMesh
{
    Size = Vector3.One
}

...

CollisionShape collisionShape = new CollisionShape()
{
	Shape = new BoxShape
	{
		Extents = 0.5f * Vector3.One
	}
};

..

Scale = new Vector3(dimensions.x, 1, dimensions.y);

```